src/core/js/models/adaptModel.js

```diff
@@ -52,7 +58,54 @@ define([
 
                     this.checkLocking();
                 }
+
+                this.setupTrackables();
+
             }, this));
+
+        },
+
+        setupTrackables: function() {
+
+            // Limit state trigger calls and make state change callbacks batched-asynchronous
+            var originalTrackableStateFunction = this.triggerTrackableState;
+            this.triggerTrackableState = _.compose(
+                _.bind(function() {
+
+                    // Flag that the function is awaiting trigger
+                    this.triggerTrackableState.isQueued = true;
+
+                }, this),
+                _.debounce(_.bind(function() {
+                    
+                    // Trigger original function
+                    originalTrackableStateFunction.apply(this);
+
+                    // Unset waiting flag
+                    this.triggerTrackableState.isQueued = false;
+
+                }, this), 17)
```

Comment:
The 17 here is a magic number.  Also, why 17?

Reply:
60fps ish. anywhere between 17ms and 40ms is usually good for stuff like this (25-60fps).
a defer would probably suffice but i wanted to group as many model changes as i could into one trigger, these number allow for a few short asynchronous model changes to occur before a callback without it seeming too laggy. > 250ms is probably too laggy.